File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/ExpenseFragmentContract.java
package com.remotearthsolutions.expensetracker.contracts;

import com.remotearthsolutions.expensetracker.databaseutils.models.AccountModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.DateModel;
import com.remotearthsolutions.expensetracker.databaseutils.models.dtos.CategoryExpense;

import java.util.List;

public interface ExpenseFragmentContract {
    interface View extends BaseView {

        void defineClickListener();

        void onExpenseAdded();

        void setSourceAccount(AccountModel account);

        void showDefaultCategory(CategoryModel categoryModel);
    }

    interface ExpenseView {

        void loadFilterExpense(List<CategoryExpense> listOffilterExpense);

        void loadDate(List<DateModel> listOfDate);
    }


}

File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/ExpenseFragmentContract.kt
package com.remotearthsolutions.expensetracker.contracts

import com.remotearthsolutions.expensetracker.databaseutils.models.AccountModel
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryExpense
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel
import com.remotearthsolutions.expensetracker.databaseutils.models.ScheduledExpenseModel

interface ExpenseFragmentContract {
    interface View : BaseView {
        fun defineClickListener()
        fun onExpenseAdded(amount: Double)
        fun onExpenseDeleted(categoryExpense: CategoryExpense?)
        fun setSourceAccount(account: AccountModel?)
        fun showDefaultCategory(categoryModel: CategoryModel?)
        fun onScheduleExpense(scheduledExpenseModel: ScheduledExpenseModel)
    }
}